src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -460,17 +453,13 @@ static Type GetMemberType(MemberInfo memberInfo)
                     JsonPropertyInfo jsonPropertyInfo = matchingEntry.JsonPropertyInfo;
                     JsonParameterInfo jsonParameterInfo = AddConstructorParameter(parameterInfo, jsonPropertyInfo, Options);
                     parameterCache.Add(jsonPropertyInfo.NameAsString, jsonParameterInfo);
-
-                    // Remove property from deserialization cache to reduce the number of JsonPropertyInfos considered during JSON matching.
-                    PropertyCache!.Remove(jsonPropertyInfo.NameAsString);
```

Comment:
You don't reckon this is beneficial?


Is there an issue if the list and dictionary don't have the exact same data (maybe/e.g. `JsonPropertyDictionary.Count` becomes ambiguous)?

Reply:
It is beneficial, but can no longer be done since there is now a single property cache contains the both list of serializable properties and deserializable properties so if we remove it the property wouldn't be serialized.

> Is there an issue if the list and dictionary don't have the exact same data (maybe/e.g. JsonPropertyDictionary.Count becomes ambiguous)?

The property cache either contains just a list or a list+dictionary and in the latter case has the same contents.